src/main/java/com/synopsys/integration/alert/provider/blackduck/tasks/BlackDuckProjectSyncTask.java

```diff
@@ -135,4 +144,32 @@ public void runTask() {
         return projectToEmailAddresses;
     }
 
+    private Set<String> retrieveAllProjectsInJobs(final Set<ProviderProject> foundProjects) {
+        final Set<String> configuredProjectNames = new HashSet<>();
+        for (final ConfigurationJobModel configurationJobModel : configurationAccessor.getAllJobs()) {
+            final CommonDistributionConfiguration commonDistributionConfiguration = new CommonDistributionConfiguration(configurationJobModel.getJobId(), configurationJobModel.createKeyToFieldMap());
+            final String projectNamePattern = commonDistributionConfiguration.getProjectNamePattern();
+            if (StringUtils.isNotBlank(projectNamePattern)) {
+                final Set<String> matchedProjectNames = foundProjects.stream().map(ProviderProject::getName).filter(projectNamePattern::matches).collect(Collectors.toSet());
+                configuredProjectNames.addAll(matchedProjectNames);
+            }
+            final Set<String> configuredProjects = commonDistributionConfiguration.getConfiguredProjects();
+            configuredProjectNames.addAll(configuredProjects);
+        }
+
+        return configuredProjectNames;
+    }
+
+    private void updateBlackDuckProjectPermissions(final Set<String> configuredProjects, final List<ProjectView> projectViews, final ProjectUsersService projectUsersService, final BlackDuckService blackDuckService)
+        throws IntegrationException {
+        final UserView currentUser = blackDuckService.getResponse(ApiDiscovery.CURRENT_USER_LINK_RESPONSE);
+        final Set<ProjectView> matchingProjects = projectViews.parallelStream()
+                                                      .filter(projectView -> configuredProjects.contains(projectView.getName()))
+                                                      .collect(Collectors.toSet());
+        for (final ProjectView projectView : matchingProjects) {
+            logger.info("Adding user to Project {}", projectView.getName());
+            projectUsersService.addUserToProject(projectView, currentUser);
```

Comment:
If the User is already assigned to the Project(s), this method will generate a log every time. We may want to wait until that is addressed in the library OR implement it ourselves.

I am in favor of using this method, fixing the library and releasing it, and update to the newer version.

Reply:
I had seen that issue when I originally implemented it. I considered doing a check before but I thought doing a check beforehand just to ignore a log was not beneficial.